letsencrypt/cli.py

```diff
@@ -642,8 +644,45 @@ def _setup_logging(args):
     logger.info("Saving debug log to %s", log_file_name)
 
 
-def main2(cli_args, args, config, plugins):
-    """Continued main script execution."""
+def _handle_exception(exc_type, exc_value, trace, args):
+    logger.debug(
+        "Exiting abnormally:\n%s",
+        "".join(traceback.format_exception(exc_type, exc_value, trace)))
+
+    if issubclass(exc_type, errors.Error) and (not args or not args.debug):
+        sys.exit(exc_value)
```

Comment:
also, I think it'd be good to `assert exc_value` to make sure than non-zero exit code is returned